### PR TITLE
Milestone 1 - Create animal table

### PR DESCRIPTION
**Pull request by Ghiftee**
This is the first part of the vet clinic database project in the Microverse Ruby/Database Module. Here I implemented the following:

- Create a database
- Create a table
- Add data to the table
- Query data from the table

Please review my code.
Thanks!

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -1,5 +1,16 @@
 /* Database schema to keep the structure of entire database. */
 
+-- CREATE TABLE animals (
+--     name varchar(100)
+-- );
+
```
Comment (arslanbisharat) [inline schema.sql]: - [x] Let's not use commented code. Leaving commented code in your codebase is a bad practice.

File: queries.sql [hunk 2]
```diff
@@ -1,3 +1,18 @@
 /*Queries that provide answers to the questions from all projects.*/
+-- SELECT * from animals WHERE name = 'Luna';
```
Comment (arslanbisharat) [inline queries.sql]: - [x] Let's remove commented code. Leaving commented code in your codebase is a bad practice.

File: data.sql [hunk 3]
```diff
@@ -1,5 +1,17 @@
 /* Populate database with sample data. */
 
-INSERT INTO animals (name) VALUES ('Luna');
-INSERT INTO animals (name) VALUES ('Daisy');
-INSERT INTO animals (name) VALUES ('Charlie');
+-- INSERT INTO animals (name) VALUES ('Luna');
+-- INSERT INTO animals (name) VALUES ('Daisy');
+-- INSERT INTO animals (name) VALUES ('Charlie');
+
```
Comment (arslanbisharat) [inline data.sql]: - [x]  Let's not use commented code. Leaving commented code in your codebase is a bad practice.

File: schema.sql [hunk 4]
```diff
@@ -1,5 +1,16 @@
...
 CREATE TABLE animals (
-    name varchar(100)
+  id INTEGER,
+  name TEXT,
+  date_of_birth DATE,
+  escape_attempts INTEGER,
+  neutered BOOLEAN,
+  weight_kg DECIMAL
 );
```
Comment (arslanbisharat) [inline schema.sql]: - [x]  Kindly add VARCHAR for all your columns as it determines the length of characters this is best practice 👍. 

- [x] [OPTIONAL] you can make the id column not null and autoincremented
`id int NOT NULL IDENTITY(1,1)`

- [x] [OPTIONAL] you can mark the id column as the primary key
`PRIMARY KEY (id)`

File: queries.sql [hunk 5]
```diff
@@ -1,3 +1,18 @@
...
 
-SELECT * from animals WHERE name = 'Luna';
+SELECT name FROM animals WHERE name LIKE '%mon';
```
Comment (arslanbisharat) [inline queries.sql]: - [x] Let's filter all animals' names and find the animals that have "mon" at the end. Currently, you are filtering only names but it's highly recommended if we filter everything such as: 
`SELECT * FROM animals WHERE name LIKE '%mon';`

File: queries.sql [hunk 6]
```diff
@@ -1,3 +1,18 @@
...
+
+SELECT * FROM animals WHERE EXTRACT(year FROM date_of_birth)BETWEEN 2016 AND 2019;
```
Comment (arslanbisharat) [inline queries.sql]: - [x] I would suggest you go through all the names of the animals and list the ones that were born between 2016 and 2019.
`SELECT name FROM animals WHERE date_of_birth >= '2016/01/01' AND date_of_birth <= '2019/31/12';`

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: schema.sql (hunk 1 above)
Comment (Ghiftee) [reply schema.sql]: Done

File: queries.sql (hunk 2 above)
Comment (Ghiftee) [reply queries.sql]: Done

File: data.sql (hunk 3 above)
Comment (Ghiftee) [reply data.sql]: Done

File: queries.sql (hunk 5 above)
Comment (Ghiftee) [reply queries.sql]: Done

File: queries.sql (hunk 6 above)
Comment (Ghiftee) [reply queries.sql]: Done

File: schema.sql (hunk 4 above)
Comment (Ghiftee) [reply schema.sql]: Done

**Comment by Ghiftee** [comment]
Thank you for the review, I have implemented all the changes
Kindly re-review my code.

**Review by AshakaE — APPROVED** [review]
### Hi @Ghiftee  🙋‍♀️, 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 4 above)
Comment (nidalaa) [reply schema.sql]: Hello @Ghiftee ,

WE own you an apologise. The dataypes of your columns were correct on your first attempt!

The main reason for this problem is that we had a mistake with the guidelines for code reviewers. We discussed it with the group of code reviewers and fixed it but a majority of the reviews that were performed before that discussion were done with the "safe mode" - by using the guidelines as a pattern.

We really apologize for the inconvenience that it caused. Good job with using correct datatypes at the beginning ❤️

We really recommend you to change them back to the initial state (in the schema.sql file but also in your database (for that please use ALET TABLE command)).

**Pull request merged**